truss/parsepkgname/parsepkgname.go

```diff
@@ -0,0 +1,111 @@
+/*
+Package parsepkgname provides functions for extracting the name of a package
+from a protocol buffer definition file. For example, given a protocol buffer 3
+file like this:
+
+	// A comment about this proto file
+	package  examplepackage;
+
+	// and the rest of the file goes here
+
+The functions in this package would extract the name "examplepackage" as the
+name of the protobuf package.
+*/
+package parsepkgname
+
+import (
+	"io"
+	"unicode"
+
+	"github.com/TuneLab/go-truss/deftree/svcparse"
+)
+
+type Token int
+
+const (
+	IDENT Token = iota
+	WHITESPACE
+	COMMENT
+	OTHER
+)
+
+type Scanner interface {
+	// ReadUnit must return groups of runes representing at least the following
+	// lexical groups:
+	//
+	//     ident
+	//     comments (c++ style single line comments and block comments)
+	//     whitespace
+	//
+	// If you need a scanner which provides these out of the box, see the
+	// SvcScanner struct in github.com/TuneLab/go-truss/deftree/svcparse
+	ReadUnit() ([]rune, error)
+}
+
+func categorize(unit []rune) Token {
+	rv := OTHER
+	r := unit[0]
+	switch {
+	case unicode.IsLetter(r):
+		rv = IDENT
+	case unicode.IsDigit(r):
+		rv = IDENT
+	case r == '_':
+		rv = IDENT
+	case unicode.IsSpace(r):
+		rv = WHITESPACE
+	case r == '/' && len(unit) > 1:
```

Comment:
I'm not entirely sure where the `unit` comes from, but just a sanity check; this is a legitimate test for a comment?